src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift

```diff
@@ -205,6 +205,17 @@ final class ExposureDetectionExecutor: ExposureDetectionDelegate {
 	}
 }
 
+/// Extension to have the device time SecureStore API available
+extension ExposureDetectionExecutor {
+	func isDeviceTimeCorrect() -> Bool {
+		return store.deviceTimeIsCorrect
+	}
+	
+	func hasDeviceTimeErrorBeenShown() -> Bool {
+		return store.deviceTimeErrorWasShown
+	}
```

Comment:
I'm really no Swift expert but I have some concerns here:
These parameters are otherwise managed by `DeviceTimeCheck`. By directly accessing the parameters from `store` here, you're making assumptions about the implementation of `DeviceTimeCheck` and couple the two tightly. Wouldn't it make more sense to access these information from the `DeviceTimeCheck` and leave if up to that class to manage all the state?

Reply:
Changed. Make use of DeviceTimeCheck to access the state.